assets/src/edit-story/components/canvas/carousel/index.js

```diff
@@ -105,6 +106,38 @@ function Carousel() {
 		} );
 	}, [ listRef ] );
 
+	const getArrangeIndex = ( sourceIndex, dstIndex, position ) => {
+		// If the dropped element is before the dropzone index then we have to deduct
+		// that from the index to make up for the "lost" element in the row.
+		const indexAdjustment = sourceIndex < dstIndex ? -1 : 0;
+		if ( 'left' === position.x ) {
+			return dstIndex + indexAdjustment;
+		}
+		return dstIndex + 1 + indexAdjustment;
+	};
+
+	const onDragStart = useCallback( ( index ) => ( evt ) => {
+		const pageData = {
+			type: 'page',
+			index,
+		};
+		evt.dataTransfer.setData( 'text', JSON.stringify( pageData ) );
+	}, [] );
+
+	const onDrop = ( evt, { position, pageIndex } ) => {
+		const droppedEl = JSON.parse( evt.dataTransfer.getData( 'text' ) );
+		if ( ! droppedEl || 'page' !== droppedEl.type ) {
+			return;
+		}
+		const arrangedIndex = getArrangeIndex( droppedEl.index, pageIndex, position );
+		// Do nothing if the index didn't change.
+		if ( droppedEl.index !== arrangedIndex ) {
+			const pageId = pages[ droppedEl.index ].id;
+			arrangePage( { pageId, position: arrangedIndex } );
+			setCurrentPage( { pageId } );
```

Comment:
I know I'm a bit late with this question, but just to confirm: does reordering of pages ever change the currently shown page?

Reply:
Yes, currently it sets the page that was reordered as current. Perhaps it shouldn't be like this? That was an assumption but maybe the current page should not change? It made more sense without the page preview that we have now.

Reply:
Opened an issue to confirm this: https://github.com/google/web-stories-wp/issues/141

Reply:
Yeah. I tried both ways and either seem pretty logical to me. I think this is low priority and let's assign to Sam.